god-mode.el

```diff
@@ -351,6 +351,76 @@ Members of the `god-exempt-major-modes' list are exempt."
            (throw 'disable t))
          (setq preds (cdr preds)))))))
 
+;;;###autoload
+(defun god-execute-with-current-bindings (&optional called-interactively)
+    "Execute a single command from God mode, preserving current keybindings.
+
+This command activates God mode temporarily, and deactivates God
+mode as soon as the next command is run.  Prefix arguments do not
+count as commands for this purpose, and do not cause God mode to
+exit.  Moreover, any prefix argument that exists at the time of
+this command's invocation is passed along to the next command.
+
+Unlike normal use of God mode, this command makes available all
+keybindings that were active at the time of its invocation,
+including keybindings that are normally invisible to God mode,
+such as those in `emulation-mode-map-alists' or text overlay
+properties.  This makes it suitable for use with packages like
+Evil that utilize such higher-priority keymaps.  (See Info
+node `(elisp)Active Keymaps' for technical details on keymap
+precedence.  For an alternative to this command, check out the
+evil-god-state package, available on MELPA.)
+
+This command has no effect when called from within God mode.
+
+For interactive use only.  CALLED-INTERACTIVELY is a dummy
+parameter to help enforce this restriction."
+    (interactive "d")
+    (if called-interactively
+        (unless god-local-mode
+          (message "Switched to God mode for the next command ...")
+          (letrec ((caller this-command)
+                   (buffer (current-buffer))
+                   (cleanup
+                    (lambda ()
+                      ;; Perform cleanup in original buffer even if the command
+                      ;; switched buffers
+                      (with-current-buffer buffer
+                        (unwind-protect (god-local-mode 0)
+                          (remove-hook 'post-command-hook post-hook)))))
+                   (kill-transient-map
+                    (set-transient-map
```

Comment:
Since `kill-transient-map` is invoked using a `funcall`, the `set-transient-map` form should be wrapped in a `lambda` form.
```emacs-lisp
(kill-transient-map
  (lambda ()
    (set-transient-map
     god-local-mode-map 'god-prefix-command-p cleanup)))
```

Reply:
I've verified this works as-is. Calling `set-transient-map` here does two things: (1) immediately activates a temporary keymap; (2) returns a dynamically-generated exit function that can be called to deactivate that keymap later. (Incidentally, the fact that `set-transient-map` returns a function is omitted from the relevant Elisp [manual page](https://www.gnu.org/software/emacs/manual/html_node/elisp/Controlling-Active-Maps.html), but the inline documentation within Emacs mentions it.)

Reply:
Whoops, you're right. I forgot `set-transient-map` returns an exit function. 👍